packages/grid/x-data-grid/src/hooks/features/columns/gridColumnsSelector.ts

```diff
@@ -66,6 +71,69 @@ export const gridVisibleColumnFieldsSelector = createSelectorMemoized(
   (visibleColumns) => visibleColumns.map((column) => column.field),
 );
 
+/**
+ * Get the visible pinned columns model.
+ * @category Visible Columns
+ */
+export const gridPinnedColumnsSelector = (state: GridStateCommunity) => state.pinnedColumns;
+
+/**
+ * Get the visible pinned columns.
+ * @category Visible Columns
+ */
+export const gridVisiblePinnedColumnDefinitionsSelector = createSelectorMemoized(
+  gridColumnsStateSelector,
+  gridPinnedColumnsSelector,
+  gridVisibleColumnFieldsSelector,
+  gridThemeSelector,
+  (columnsState, model, visibleColumnFields, theme) => {
+    const visiblePinnedFields = filterVisibleColumns(
+      model,
+      visibleColumnFields,
+      theme.direction === 'rtl',
```

Comment:
Should we go a step further here and create a `gridThemeDirectionSelector` to preserve caching when the theme changes, but the direction is the same?

Reply:
I don't expect the theme object to change often enough to be worth it.